nni/algorithms/compression/pytorch/pruning/admm_pruner.py

```diff
@@ -59,12 +59,14 @@ def trainer(model, criterion, optimizer, epoch, callback):
 
     """
 
-    def __init__(self, model, config_list, trainer, num_iterations=30, training_epochs=5, row=1e-4, base_algo='l1'):
+    def __init__(self, model, config_list, trainer, criterion, optimizer=None, num_iterations=30, training_epochs=5, row=1e-4, base_algo='l1'):
```

Comment:
Why increase two arguments?

Reply:
How about giving `criterion` a default value (torch.nn.CrossEntropyLoss)

Reply:
Thanks! AdmmPruner is moved into `iterative_pruner.py`. Two arguments are aligned with the parameters of `IterativePruner`. The default values are added for parameters, please review the latest version.